### PR TITLE
feat: functional upgrade-10 handler

### LINKED ISSUE #7357: Upgrade handler for upgrade-10 (vaults)
Need an upgrade handler for the next release.

### PR DESCRIPTION
**Pull request by JimLarson**
closes: #7357

## Description

Runs the vstorage migration function in the placeholder.

Implicitly runs per-module migration for `x/swingset` to add new beans for storage (see #7290).

### Security Considerations

N/A

### Scaling Considerations

N/A

### Documentation Considerations

N/A

### Testing Considerations

Need to have an upgrade end-to-end test as part of release checklist.

### CODE REVIEW

File: golang/cosmos/app/app.go [hunk 1]
```diff
@@ -751,6 +751,22 @@ func NewAgoricApp(
 		upgrade10Handler(app, upgradeNameTest),
 	)
 
+	/* REVIEWER: confirm that this isn't needed
```
Comment (mhofman) [inline golang/cosmos/app/app.go]: I have no idea what this is supposed to be about, so I'll let other reviewers review.

File: golang/cosmos/app/app.go [hunk 2]
```diff
@@ -773,8 +789,7 @@ func NewAgoricApp(
 
 func upgrade10Handler(app *GaiaApp, targetUpgrade string) func(sdk.Context, upgradetypes.Plan, module.VersionMap) (module.VersionMap, error) {
 	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVm module.VersionMap) (module.VersionMap, error) {
-		// TODO: In next upgrade handler (bulldozer), run:
-		// app.VstorageKeeper.MigrateNoDataPlaceholders(ctx)
+		app.VstorageKeeper.MigrateNoDataPlaceholders(ctx)
```
Comment (mhofman) [inline golang/cosmos/app/app.go]: Maybe add a comment to mention this should only be run in Bulldozer, and not in future upgrades (aka do not include in upgrade11).

Also we'll need a staging environment equivalent I suppose?

**Review by mhofman — COMMENTED** [review]
I cannot review the commented out bit, so I'm deferring to another reviewer.

File: golang/cosmos/app/app.go (hunk 2 above)
Comment (JimLarson) [reply golang/cosmos/app/app.go]: Added a note, though I think it's implicit that the contents of a specific upgrade handler are just for that specific upgrade.

The staging upgrade should be handled by the `upgradeNameTest` handler installed on line 749.

File: golang/cosmos/app/app.go (hunk 1 above)
Comment (arirubinstein) [reply golang/cosmos/app/app.go]: if this is aligned to the rho there were no store changes and we can skip this part

Comment (JimLarson) [reply golang/cosmos/app/app.go]: No, not rho. We're still on cosmos-sdk 0.45.x. But the only change from upgrade-9 (pismoC) are the state sync changes which I believe don't need upgrade support. Is that correct, @mhofman ?

Comment (mhofman) [reply golang/cosmos/app/app.go]: Correct, state-sync is a new feature so it doesn't need any migration.

**Comment by JimLarson** [comment]
I seek approval, and for the PR too.

**Review by mhofman — APPROVED** [review]

**Pull request merged**